src/generics.md

```diff
@@ -113,33 +114,32 @@ This example has a few different substitutions:
 
 Let’s look a bit more closely at that last substitution to see why we use indexes. If we want to
 find the type of `foo.x`, we can get generic type of `x`, which is `Vec<Param(0)>`. Now we can take
-the index `0` and use it to find the right type substitution: looking at `Foo`'s `SubstsRef`, we
-have the list `[u32, f32]` , since we want to replace index `0`, we take the 0-th index of this
+the index `0` and use it to find the right type substitution: looking at `Foo`'s `GenericArgsRef`,
+we have the list `[u32, f32]` , since we want to replace index `0`, we take the 0-th index of this
 list, which is `u32`. Voila!
 
 You may have a couple of followup questions…
 
- **`type_of`** How do we get the “generic type of `x`"? You can get the type of pretty much anything
+ **`type_of`** How do we get the "generic type of `x`"? You can get the type of pretty much anything
  with the   `tcx.type_of(def_id)` query. In this case, we would pass the `DefId` of the field `x`.
  The `type_of` query always returns the definition with the generics that are in scope of the
  definition. For example, `tcx.type_of(def_id_of_my_struct)` would return the “self-view” of
  `MyStruct`: `Adt(Foo, &[Param(0), Param(1)])`.
 
-**`subst`** How do we actually do the substitutions? There is a function for that too! You use
-[`subst`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/subst/struct.EarlyBinder.html#method.subst) to
-replace a `SubstsRef` with another list of types.
+**`instantiate`** How do we actually do the substitutions? There is a function for that too! You
+use [`instantiate`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/
+generic_args/struct.EarlyBinder.html#method.instantiate) to replace a `GenericArgsRef` with
+another list of types.
 
-[Here is an example of actually using `subst` in the compiler][substex].  The exact details are not
-too important, but in this piece of code, we happen to be converting from the `rustc_hir::Ty` to
-a real `ty::Ty`. You can see that we first get some substitutions (`substs`).  Then we call
-`type_of` to get a type and call `ty.subst(substs)` to get a new version of `ty` with
-the substitutions made.
+[Here is an example of actually using `instantiate` in the compiler][instantiatex].
+The exact details are not too important, but in this piece of code, we happen to be
+converting from the `rustc_hir::Ty` to a real `ty::Ty`. You can see that we first get some args
+(`args`).  Then we call `type_of` to get a type and call `ty.instantiate(tcx, args)` to get a new
+version of `ty` with the args made.
 
-[substex]: https://github.com/rust-lang/rust/blob/0940040c0486a536be4f8685c7dd9a078f9e87c2/compiler/rustc_hir_analysis/src/astconv/mod.rs#L1231-L1242
+[instantiatex]: https://github.com/rust-lang/rust/blob/master/compiler/rustc_hir_analysis/src/astconv/mod.rs#L905-L927
```

Comment:
pinning to a commit to avoid link rot
```suggestion
[instantiatex]: https://github.com/rust-lang/rust/blob/[master](https://github.com/rust-lang/rust/blob/8a562f9671e36cf29c9c794c2646bcf252d55535/compiler/rustc_hir_analysis/src/astconv/mod.rs#L905-L927)/compiler/rustc_hir_analysis/src/astconv/mod.rs#L905-L927
```